exchanges/kraken/kraken_websocket.go

```diff
@@ -210,280 +196,257 @@ func isAwaitingCancelOrderResponses(requestID int64, success bool) bool {
-					Exchange: k.Name,
-					OrderID:  status.TransactionID,
-					Status:   order.New,
-				}
+		k.Websocket.DataHandler <- &order.Detail{
+			Exchange: k.Name,
+			OrderID:  status.TransactionID,
+			Status:   order.New,
+		}
 
-				if !isChannelExist && status.RequestID > 0 {
-					return fmt.Errorf("can't send ws incoming data to Matched channel with RequestID: %d",
-						status.RequestID)
-				}
-			case krakenWsSubscriptionStatus:
-				var sub wsSubscription
-				err := json.Unmarshal(respRaw, &sub)
-				if err != nil {
-					return fmt.Errorf("%s - err %s unable to parse subscription response: %s",
-						k.Name,
-						err,
-						respRaw)
-				}
-				if sub.Status != "subscribed" && sub.Status != "unsubscribed" {
-					return fmt.Errorf("%v %v %v",
-						k.Name,
-						sub.RequestID,
-						sub.ErrorMessage)
-				}
-				k.addNewSubscriptionChannelData(&sub)
-				if sub.RequestID > 0 {
-					k.Websocket.Match.IncomingWithData(sub.RequestID, respRaw)
-				}
-			default:
-				k.Websocket.DataHandler <- stream.UnhandledMessageWarning{
-					Message: k.Name + stream.UnhandledMessage + string(respRaw),
-				}
-			}
-			return nil
+		if !isChannelExist && status.RequestID > 0 {
+			return fmt.Errorf("can't send ws incoming data to Matched channel with RequestID: %d",
+				status.RequestID)
 		}
-	}
-	return nil
-}
+	case krakenWsSubscriptionStatus:
+		var sub wsSubscription
+		err := json.Unmarshal(respRaw, &sub)
+		if err != nil {
+			return fmt.Errorf("%s - err %s unable to parse subscription response: %s",
+				k.Name,
+				err,
+				respRaw)
+		}
+		if sub.RequestID == 0 {
+			return fmt.Errorf("%v %w: %v", k.Name, errNoRequestID, respRaw)
+		}
+		k.Websocket.Match.IncomingWithData(sub.RequestID, respRaw)
 
-// wsPingHandler sends a message "ping" every 27 to maintain the connection to the websocket
-func (k *Kraken) wsPingHandler() error {
-	message, err := json.Marshal(pingRequest)
-	if err != nil {
-		return err
+		if sub.Status != "subscribed" && sub.Status != "unsubscribed" {
+			return fmt.Errorf("%v %v %v",
+				k.Name,
+				sub.RequestID,
+				sub.ErrorMessage)
+		}
+	default:
+		k.Websocket.DataHandler <- stream.UnhandledMessageWarning{
+			Message: k.Name + stream.UnhandledMessage + string(respRaw),
+		}
 	}
-	k.Websocket.Conn.SetupPingHandler(stream.PingHandler{
-		Message:     message,
-		Delay:       krakenWsPingDelay,
-		MessageType: websocket.TextMessage,
-	})
+
 	return nil
 }
 
-// wsAuthPingHandler sends a message "ping" every 27 to maintain the connection to the websocket
-func (k *Kraken) wsAuthPingHandler() error {
-	message, err := json.Marshal(pingRequest)
-	if err != nil {
-		return err
-	}
-	k.Websocket.AuthConn.SetupPingHandler(stream.PingHandler{
-		Message:     message,
+// wsPingHandler starts a websocket ping handler every 27s
+func (k *Kraken) wsPingHandler(conn stream.Connection) {
+	conn.SetupPingHandler(stream.PingHandler{
+		Message:     []byte(`{"event":"ping"}`),
```

Comment:
Really didn't see the point of a fancy ping handler for this. It just feels obtuse and inefficient.